### PR TITLE
Fix for missing port in redirect to https

### PR DESCRIPTION
**Pull request by mattjackson220**
## What does this PR (Pull Request) do?
Fixes a bug in the TM HTTPS code that was preventing redirects from working between the HTTP endpoint and the new HTTPS endpoint.

- [X] This PR is not related to any Issue 

## Which Traffic Control components are affected by this PR?
- Traffic Monitor
- Traffic Router
No documentation because it is a minor fix and the big change has already been documented
No tests included for same reason

## What is the best way to verify this PR?
In CDN In A Box, update the TM run script with these properties:
"httpsListener": ":443",
"certFile": "/shared/ssl/infra.ciab.test.crt",
"keyFile": "/shared/ssl/infra.ciab.test.key"

Verify that the TM HTTP endpoint redirects to the HTTPS endpoint and the the HTTPS endpoint shows the expected information.  Verify that Traffic Router successfully gets the configuration from TM through the redirect.  Verify that Traffic Stats successfully calls TM through redirect.

- [X] This PR includes tests OR I have explained why tests are unnecessary
- [X] This PR includes documentation OR I have explained why documentation is unnecessary
- [X] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [X] This PR includes any and all required license headers
- [X] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [X] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by dangogh** [comment]
add to whitelist

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3761/
Test PASSed.

File: traffic_monitor/srvhttp/srvhttp.go [hunk 1]
```diff
@@ -186,7 +187,15 @@ func (s *Server) RunHTTPSRedirect(addr string, addrForRedirect string, readTimeo
...
 
 func (s *Server) redirectTLS(w http.ResponseWriter, r *http.Request) {
-	host, _, _ := net.SplitHostPort(r.Host)
+	host, _, err := net.SplitHostPort(r.Host)
+	if err != nil {
+		if strings.Contains(err.Error(), "missing port in address") {
+			host = r.Host
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(`{"error": "marshalling: ` + err.Error() + `"}`))
```
[the remark this answers is not in the document]
Comment (rob05c) [reply traffic_monitor/srvhttp/srvhttp.go]: Yeah, this is writing the `500` code, but then it goes ahead and calls `http.Redirect` below. Is this supposed to have a `return` here?

[the remark this answers is not in the document]
Comment (mattjackson220) [reply traffic_monitor/srvhttp/srvhttp.go]: done. updated the error message and added the return

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/3888/
Test PASSed.

**Pull request merged**